### ISSUE TITLE
All controls in profile evaluated when controls are selected using --controls

### ISSUE THREAD

**Issue by rx294** [inspec/inspec#2857]
### 🎛 Description
Inspec cli allows for selection of controls using the following tag
```
--controls=one two three
A list of controls to run. Ignore all other tests.
```
However inspec runner still evals all the controls after which the selected controls filtered and poulated in the results.

This causes the runtime to be identical whether the entire profile is run or a selected few controls.
Inspec also goes through runtime errors if any of the unselected controls causes an exception.
 
This is a significant issue when working with very large profiles like RHEL7.

**Choose one:** is this a 🐛 bug report 

Briefly describe the issue and the expected behavior

### 🌍 InSpec and Platform Version

2.1.0

Aha! Link: https://chef.aha.io/features/SH-468

**Comment by aaronlippold** [comment]
I think this also gives us insight into how we would look at doing - https://github.com/chef/inspec/issues/1677

**Comment by jquick** [comment]
This is something we are looking at fixing in a bigger overhaul soon. I will link the ticket once its flushed out.

**Comment by chris-rock** [comment]
This would be nice 👍

**Comment by aaronlippold** [comment]
this needs to be put back on the worksheet :)

**Comment by samcornwell** [comment]
Looking at tackling this issue. I think I've figured out how to filter controls before they are executed, but I wondered if you need me to remove the current post-eval filtering that is in the code before you'd accept a PR. It would be redundant to have the results filtering remain, but theoretically not breaking anything.

**Comment by aaronlippold** [comment]
bump

**Comment by clintoncwolfe** [comment]
This was fixed on #5434
